File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-fx-5-4/FxMultipleExercise.cpp
// FxMultipleExercise.cpp 
#include "FxMultipleExercise.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

FxMultipleExercise::FxMultipleExercise(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //minimumNotionalAmountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* minimumNotionalAmountNode = xmlNode->FirstChildElement("minimumNotionalAmount");

   if(minimumNotionalAmountNode){minimumNotionalAmountIsNull_ = false;}
   else{minimumNotionalAmountIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- minimumNotionalAmountNode , address : " << minimumNotionalAmountNode << std::endl;
   #endif
   if(minimumNotionalAmountNode)
   {
      if(minimumNotionalAmountNode->Attribute("href") || minimumNotionalAmountNode->Attribute("id"))
      {
          if(minimumNotionalAmountNode->Attribute("id"))
          {
             minimumNotionalAmountIDRef_ = minimumNotionalAmountNode->Attribute("id");
             minimumNotionalAmount_ = boost::shared_ptr<NonNegativeMoney>(new NonNegativeMoney(minimumNotionalAmountNode));
             minimumNotionalAmount_->setID(minimumNotionalAmountIDRef_);
             IDManager::instance().setID(minimumNotionalAmountIDRef_,minimumNotionalAmount_);
          }
          else if(minimumNotionalAmountNode->Attribute("href")) { minimumNotionalAmountIDRef_ = minimumNotionalAmountNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { minimumNotionalAmount_ = boost::shared_ptr<NonNegativeMoney>(new NonNegativeMoney(minimumNotionalAmountNode));}
   }

   //maximumNotionalAmountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* maximumNotionalAmountNode = xmlNode->FirstChildElement("maximumNotionalAmount");

   if(maximumNotionalAmountNode){maximumNotionalAmountIsNull_ = false;}
   else{maximumNotionalAmountIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- maximumNotionalAmountNode , address : " << maximumNotionalAmountNode << std::endl;
   #endif
   if(maximumNotionalAmountNode)
   {
      if(maximumNotionalAmountNode->Attribute("href") || maximumNotionalAmountNode->Attribute("id"))
      {
          if(maximumNotionalAmountNode->Attribute("id"))
          {
             maximumNotionalAmountIDRef_ = maximumNotionalAmountNode->Attribute("id");
             maximumNotionalAmount_ = boost::shared_ptr<NonNegativeMoney>(new NonNegativeMoney(maximumNotionalAmountNode));
             maximumNotionalAmount_->setID(maximumNotionalAmountIDRef_);
             IDManager::instance().setID(maximumNotionalAmountIDRef_,maximumNotionalAmount_);
          }
          else if(maximumNotionalAmountNode->Attribute("href")) { maximumNotionalAmountIDRef_ = maximumNotionalAmountNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { maximumNotionalAmount_ = boost::shared_ptr<NonNegativeMoney>(new NonNegativeMoney(maximumNotionalAmountNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<NonNegativeMoney> FxMultipleExercise::getMinimumNotionalAmount()
{
   if(!this->minimumNotionalAmountIsNull_){
        if(minimumNotionalAmountIDRef_ != ""){
             return boost::shared_static_cast<NonNegativeMoney>(IDManager::instance().getID(minimumNotionalAmountIDRef_));
        }else{
             return this->minimumNotionalAmount_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<NonNegativeMoney>();
   }
}
boost::shared_ptr<NonNegativeMoney> FxMultipleExercise::getMaximumNotionalAmount()
{
   if(!this->maximumNotionalAmountIsNull_){
        if(maximumNotionalAmountIDRef_ != ""){
             return boost::shared_static_cast<NonNegativeMoney>(IDManager::instance().getID(maximumNotionalAmountIDRef_));
        }else{
             return this->maximumNotionalAmount_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<NonNegativeMoney>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-fx-5-4/FxMultipleExercise.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class FxMultipleExercise : ISerialized
    {
        public FxMultipleExercise(XmlNode xmlNode)
        {
            XmlNode minimumNotionalAmountNode = xmlNode.SelectSingleNode("minimumNotionalAmount");
            
            if (minimumNotionalAmountNode != null)
            {
                if (minimumNotionalAmountNode.Attributes["href"] != null || minimumNotionalAmountNode.Attributes["id"] != null) 
                {
                    if (minimumNotionalAmountNode.Attributes["id"] != null) 
                    {
                        minimumNotionalAmountIDRef_ = minimumNotionalAmountNode.Attributes["id"].Value;
                        NonNegativeMoney ob = new NonNegativeMoney(minimumNotionalAmountNode);
                        IDManager.SetID(minimumNotionalAmountIDRef_, ob);
                    }
                    else if (minimumNotionalAmountNode.Attributes["href"] != null)
                    {
                        minimumNotionalAmountIDRef_ = minimumNotionalAmountNode.Attributes["href"].Value;
                    }
                    else
                    {
                        minimumNotionalAmount_ = new NonNegativeMoney(minimumNotionalAmountNode);
                    }
                }
                else
                {
                    minimumNotionalAmount_ = new NonNegativeMoney(minimumNotionalAmountNode);
                }
            }
            
        
            XmlNode maximumNotionalAmountNode = xmlNode.SelectSingleNode("maximumNotionalAmount");
            
            if (maximumNotionalAmountNode != null)
            {
                if (maximumNotionalAmountNode.Attributes["href"] != null || maximumNotionalAmountNode.Attributes["id"] != null) 
                {
                    if (maximumNotionalAmountNode.Attributes["id"] != null) 
                    {
                        maximumNotionalAmountIDRef_ = maximumNotionalAmountNode.Attributes["id"].Value;
                        NonNegativeMoney ob = new NonNegativeMoney(maximumNotionalAmountNode);
                        IDManager.SetID(maximumNotionalAmountIDRef_, ob);
                    }
                    else if (maximumNotionalAmountNode.Attributes["href"] != null)
                    {
                        maximumNotionalAmountIDRef_ = maximumNotionalAmountNode.Attributes["href"].Value;
                    }
                    else
                    {
                        maximumNotionalAmount_ = new NonNegativeMoney(maximumNotionalAmountNode);
                    }
                }
                else
                {
                    maximumNotionalAmount_ = new NonNegativeMoney(maximumNotionalAmountNode);
                }
            }
            
        
        }
        
    
        #region MinimumNotionalAmount_
        private NonNegativeMoney minimumNotionalAmount_;
        public NonNegativeMoney MinimumNotionalAmount_
        {
            get
            {
                if (this.minimumNotionalAmount_ != null)
                {
                    return this.minimumNotionalAmount_; 
                }
                else if (this.minimumNotionalAmountIDRef_ != null)
                {
                    minimumNotionalAmount_ = IDManager.getID(minimumNotionalAmountIDRef_) as NonNegativeMoney;
                    return this.minimumNotionalAmount_; 
                }
                else
                {
                      return this.minimumNotionalAmount_; 
                }
            }
            set
            {
                if (this.minimumNotionalAmount_ != value)
                {
                    this.minimumNotionalAmount_ = value;
                }
            }
        }
        #endregion
        
        public string minimumNotionalAmountIDRef_ { get; set; }
        #region MaximumNotionalAmount_
        private NonNegativeMoney maximumNotionalAmount_;
        public NonNegativeMoney MaximumNotionalAmount_
        {
            get
            {
                if (this.maximumNotionalAmount_ != null)
                {
                    return this.maximumNotionalAmount_; 
                }
                else if (this.maximumNotionalAmountIDRef_ != null)
                {
                    maximumNotionalAmount_ = IDManager.getID(maximumNotionalAmountIDRef_) as NonNegativeMoney;
                    return this.maximumNotionalAmount_; 
                }
                else
                {
                      return this.maximumNotionalAmount_; 
                }
            }
            set
            {
                if (this.maximumNotionalAmount_ != value)
                {
                    this.maximumNotionalAmount_ = value;
                }
            }
        }
        #endregion
        
        public string maximumNotionalAmountIDRef_ { get; set; }
        
    
        
    
    }
    
}

